editor/scene/2d/tiles/tiles_editor_plugin.cpp

```diff
@@ -145,13 +162,34 @@ void TilesEditorUtils::_thread() {
 void TilesEditorUtils::queue_pattern_preview(Ref<TileSet> p_tile_set, Ref<TileMapPattern> p_pattern, Callable p_callback) {
 	ERR_FAIL_COND(p_tile_set.is_null());
 	ERR_FAIL_COND(p_pattern.is_null());
+
+	// Generate cache key based on pattern content and tileset
+	String cache_key = String::num_int64(p_tile_set->get_instance_id()) + "_" + String::num_int64(p_pattern->get_instance_id());
```

Comment:
I am not sure this it enough. The problem is that, if the content of the tileset is modifed, then all stored preview will simply be outdated. It likely needs to be fully cleared if the TileSet is modified.